liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1277,7 +1277,14 @@ private void doDevMode() throws MojoExecutionException {
         List<MavenProject> upstreamMavenProjects = new ArrayList<MavenProject>();
         ProjectDependencyGraph graph = session.getProjectDependencyGraph();
         if (graph != null) {
-            checkMultiModuleConflicts(graph);
+        	
+        	// In a multi-module build, dev mode will only be run on one project (the farthest downstream) and compile will
+        	// be run on any relative upstream projects. If this current project in the Maven Reactor is not one of those projects, skip it.  
+        	List<MavenProject> relevantProjects = getRelevantMultiModuleProjects(graph);
+        	if (!relevantProjects.contains(project)) {
+        		getLog().info("\nSkipping module " + project.getArtifactId() + " which is not configured for dev mode.\n");
```

Comment:
This ends up being a misleading message.  

If the current project is configured to be skipped, we won't get this far.   We'll have already skipped back in the first check we do for skip being set in `io.openliberty.tools.maven.server.DevMojo.execute()`.

I think we only will get here if we skip because it's a JAR module.   Maybe we should just say that. 

I think a debug msg could be forgiven for being less than perfect since it's more for the plugin dev team / us ... but I think an info msg should not contribute to confusion.... "not configured for dev mode" sounds like "skip=true" to me.

Reply:
As you look at how to fix this... one idea would be to move the logging into the "getRelevantMultiModuleProjects" method (it already has debug logging).   

You could pass the goal name and achieve some small amt. of code reuse with something more like this (what the heck, I renamed the method too:)

```java
        boolean applyMultiModuleFilter(graph, project, goalName)		  
		
          if (!applyMultiModuleFilter(graph, project, goalName) {
		     // return
```

I wouldn't have mentioned this purely for code reuse :) but like I was saying, the msg concerned me.

Reply:
Yea this message was giving Cheryl and I some grief too before we settled on this one.  Like you said, skip=true will result in a skip in the execute() method, but for this new check, we wont necessarily only skip on jar projects. 
1. jar projects that are upstream projects of the "selected" downstream project wont be skipped.
2. jar projects that are _not_ leaf projects could also be skipped here (e.g. a flow of jar2 -> jar3)
3. war projects could also be skipped here (e.g. a flow of jar2 -> war2 -> ear2 where "skip" is configured on the ear2 project).

We could change to "which is not included in this dev mode execution" ?

Reply:
I think I see what you're saying...   

In 3., we'd skip a WAR even though skip was not explicitly configured on the WAR, because the WAR was upstream of a leaf that was configured to skip.

So "configured for dev mode" might be indirect.     

Yeah I think "not included in this...execution" is a bit better.  Thx for explaining.

Reply:
Are we allowed to say "execution"? I thought that was a no no considering the other meanings of the word.

Reply:
"invocation" better?